Update grafana content to reflect current practices

idavidmcdonald opened the pull request:
## Why
https://trello.com/c/CxkcgEk2/453-update-our-grafana-documentation-for-our-users
- It is rare that teams are using the GDS default dashboard because
they are now creating their own dashboards in their team folders and using those
- We also have other sources of metrics meaning teams may also
use the container metrics dashboard or dashboards for tools like
elasticsearch.

## What
- Added a short explanation for how team folders are organised
- Added definitions of the purpose of the dashboards in the
General folder
- Moved the detailed explanation of the default application dashboard
into it's own page (but have made no content changes)
- Removed some of the simpler content as I don't think we have seen
evidence of users struggling to use basic Grafana functionality so far

## Approach
- I have kept the section on team dashboard permissions brief
and vague - essentially contact us for help - as I think we need
to do more on team permissions. At the moment, teams will not
be able to manage the permissions for users of their team without
having admin privileges of the whole system as per
https://github.com/grafana/grafana/issues/11584. This may be something
we want to investigate further to avoid us needing to manage all
user permissions in terms of which teams they can access and instead
being able to put this burden on the teams themselves.
- I moved the detailed explanation of the default GDS application dashboard to avoid
cluttering the overview of grafana page too much. It is also questionable
how useful this page is for our users (my hypothesis is that it is not)
but I am not confident enough to just remove it so have instead moved it.

kentsanggds committed:
Update grafana content to reflect current practices

- It is rare that teams are using the GDS default dashboard because
they are now creating their own dashboards and using those
- We also have other sources of metrics meaning teams may also
use the container metrics dashboard or metrics for tools like
elasticsearch.

Based on this I have:

- Added a short explanation for how team folders are organised
- Added definitions of the purpose of the dashboards in the
General folder
- Moved the detailed explanation of the default application dashboard
into it's own page (but have made no content changes)
- Removed some of the simpler content as I don't think we have seen
evidence of users struggling to use Grafana so far

Note: I have kept the section on team dashboard permissions brief
and vague - essentially contact us for help - as I think we need
to do more on team permissions. At the moment, teams will not
be able to manage the permissions for users of their team without
having admin privileges of the whole system as per
https://github.com/grafana/grafana/issues/11584. This may be something
we want to investigate further to avoid us needing to manage all
user permissions in terms of which teams they can access and instead
being able to put this burden on the teams themselves.

ejrowley reviewed (approved):
An improvement over the existing docs

soupdragon99 committed:
Update index.html.md.erb

soupdragon99 committed:
Update index.html.md.erb

soupdragon99 committed:
Tech writer edits - first pass

Edits to match GOV.UK house stye and layout guidelines.

soupdragon99 commented on source/manuals/grafana-dashboards.html.md.erb:
```diff
@@ -1,58 +1,61 @@
+Contact Reliability Engineering using the [#re-prometheus-support Slack channel][https://gds.slack.com/messages/CAF5H4N4Q/#] to request:
 
+* admin permissions to create or edit a dashboard in your team folder
+* a new team folder and the admin rights to manage it
```
Is there a naming convention people should follow for team folders?

idavidmcdonald replied:
It's still kind of early days with the team folders so there aren't any super precise rules at the moment

soupdragon99 commented on source/manuals/grafana-dashboards.html.md.erb:
```diff
@@ -1,58 +1,61 @@
 ## Default and example dashboards
 
-In the 'General' dashboard folder, we provide default and template dashboards for you to use. These are a good starting point for customisation or if you have just set up monitoring to check that everything works as expected.
+You can use the default and template dashboards in the **General** dashboard folder. These are a good starting point for customisation or if you have set up monitoring to check everything works as expected.
```
What's the difference between default and template dashboards, we could say what they are.

idavidmcdonald replied:
@soupdragon99 
Good question.

We currently have four dashboards in the General folder
- GDS Application Metrics Default Dashboard
- GDS Container Metrics Default Dashboard
- Postgres
- MySQL

The first two are dashboards that we originally thought teams would use for displaying their metrics but it seems to have diverged to them now creating their own dashboard in a team folder that may be based on the default dashboards. (I don't know if people still do use the Default dashboard as their only source of viewing metrics for their apps). They were made by our team to show what we think is best practice for viewing metrics from your applications or containers.

The Postgres and MySQL dashboards I believe are downloaded from the community dashboards and could be used by a team if they started to export Postgres or MySQL metrics. I don't think anyone is using them at the moment but we haven't deleted them.

I hope that gives you some context about why I have been thinking of the first two dashboards as 'default' and the second two as 'examples'. Saying that, we could just call them all 'default' or all 'example' dashboards to simplify things. I think 'example' might be a more descriptive word so would suggest that. Any thoughts?

soupdragon99 commented on source/manuals/grafana-dashboards.html.md.erb:
```diff
@@ -1,58 +1,61 @@
 
 ### Official and community built dashboards
 
 You can [import dashboards](http://docs.grafana.org/reference/export_import/#importing-a-dashboard) from the [official and community built dashboards](https://grafana.com/dashboards) on the Grafana website.
```
It would be good to say what the benefits of this would be and if there's any we think could be particularly useful to people.

soupdragon99 replied:
It would be good to say what the benefits of this would be and if there's any we think could be particularly useful to people.

soupdragon99 commented:
@idavidmcdonald I have updated the PR with some edits and a few questions\suggestions.

My other thought was to just remove the detailed explanation of the default GDS application dashboard page as part of this PR. It would be a decisive way to test your hypothesis and it could always be added back easily enough if it's something people still want.

idavidmcdonald committed:
Edits based on comments

soupdragon99 committed:
Update grafana-dashboards.html.md.erb

idavidmcdonald committed:
Small tweak to community built dashboards

I felt the previous sentence implied that only dashboards
for backing services were provided but those are just some of
the more common uses.

idavidmcdonald committed:
Remove explanation of GDS default dashboard

Our hypothesis is that our users are not using this and do not
find it useful. If we find evidence to add it back it then we can
do so.

soupdragon99 committed:
minor edits

matching guidance title closer to the content, removing passive voice, adding detail to the Display Dashboards section. Removed guidance intro and first heading - it just needs to tell people 'how to do the thing'

soupdragon99 committed:
Merge branch 'update-grafana-docs' of https://github.com/alphagov/reliability-engineering into update-grafana-docs

soupdragon99 committed:
Update landing page linnk to match guidance

soupdragon99 committed:
Peer review edits

soupdragon99 commented:
@idavidmcdonald I have updated the guidance based on feedback from 2i peer review, these suggestions are based on the house style and things we find work with other guidance. These are mostly small changes.

I do think we should remove the remaining three screen images from the guidance. I think what we have written is clear and the Grafana interface is easy to navigate. It also means there's no overhead keeping images up to date, for example, the login screen has already changed from the current image.

Thanks,
Pete

soupdragon99 committed:
Update grafana-dashboards.html.md.erb
